src/utils/unique_email_enforcement/storage.ts

```diff
@@ -0,0 +1,58 @@
+import { TableClient, odata, TableEntityResult } from "@azure/data-tables";
+import { EmailString } from "@pagopa/ts-commons/lib/strings";
+
+import { ProfileEmail, ProfileEmailReader, ProfileEmailWriter } from "./index";
+
+// Generates AsyncIterable<ProfileEmail> from AsyncIterable<TableEntityResult>
+export async function* toProfileEmailsAsyncIterator(
+  iterator: AsyncIterableIterator<TableEntityResult<unknown>>
+): AsyncIterableIterator<ProfileEmail> {
+  for await (const item of iterator) {
+    const uniqueEmail = ProfileEmail.decode({
+      email: item.partitionKey,
+      fiscalCode: item.rowKey
+    });
+    if (uniqueEmail._tag === "Right") {
+      yield uniqueEmail.right;
+    }
```

Comment:
Invalid values are excluded from the stream, do you think that it better raise an error in this case? :) 
(I don't used `E.isRight` to avoid dependencies in `fp-ts` for this module, since that is not used in other parts)

Reply:
(added `E.isRight` !)

Reply:
> Invalid values are excluded from the stream, do you think that it better raise an error in this case? :)

We are decoding items retrieved from a query based on email. Said that it should never happen, if we strip away values we could return wrong information (an email is not taken while it has been stored).

Reply:
done!